sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1016,17 +1010,11 @@ class Column(val expr: Expression) extends Logging {
    *   df.select($"colA".name("colB"))
    * }}}
    *
-   * If the current column has metadata associated with it, this metadata will be propagated
-   * to the new column.  If this not desired, use `as` with explicitly empty metadata.
```

Comment:
Anybody can tell me why we need to remove these comments?

Reply:
@gatorsmile . From my understanding, it was because this is `inherited` from the child instead of `propagating`.

Reply:
I think we should keep them. The original comments are still valid. If users want to block metadata propagation, they should call the API `as(alias: String, metadata: Metadata):` with an empty metadata.

Reply:
I'd say the comment is slightly invalid as of now, depending on how we interpret "propagation" here. If someone interprets propagation as a "copy" then that becomes wrong. It's more likely that we leave it on the behavior of Alias without touching.

And there're overloaded method which receives metadata, which may get rid of necessity of explaining.

But yes that doesn't mean we need to remove these comments - that's also an option we may reword a bit to try resolving ambiguity and keep leaving the part of comment instead.

Reply:
To me, it sounds like they are the same to our end users. 

I do not have a strong opinion on this. Feel free to rewrite it. When writing the user-facing comments, please avoid explaining the implementation details. We just need to document the external impact and behaviors.

Reply:
I'll recover them in this PR.